Pixel 6a / Android 12 crashes when trying to open OpenHab

### Actual behaviour
The app immediately crashes on my pixel 6a (running Android 12) whenever I try to open it. This is a new phone with the latest version of openHab installed from the play store (2.20.0)


### Expected behaviour
The app should open normally.

 
### Steps to reproduce
1. Download the app on a pixel 6a.
2. Attempt to run the app and see that the app won't open.


### Can you reproduce the issue in demo mode?
I can't get into demo model.


### Environment data
#### Client
* Android version: 12
* Device model: Pixel 6a
* App version 2.20.0
* Build flavor <!-- Play Store and GitHub = "Full", F-Droid = "Foss". If self-built and you haven't changed the flavor, it's Foss -->: Downloaded from play store
* Device language: English

For this combination of openHAB app version, device and Android version I see a single crash:

````
java.lang.RuntimeException: 
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:6767)
  at android.app.ActivityThread.access$1500 (ActivityThread.java:256)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2091)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loopOnce (Looper.java:201)
  at android.os.Looper.loop (Looper.java:288)
  at android.app.ActivityThread.main (ActivityThread.java:7870)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:548)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1003)
Caused by: com.google.crypto.tink.shaded.protobuf.InvalidProtocolBufferException: 
  at com.google.crypto.tink.shaded.protobuf.GeneratedMessageLite.parsePartialFrom (GeneratedMessageLite.java:1566)
  at com.google.crypto.tink.shaded.protobuf.GeneratedMessageLite.parseFrom (GeneratedMessageLite.java:1664)
  at com.google.crypto.tink.proto.Keyset.parseFrom (Keyset.java:957)
  at com.google.crypto.tink.integration.android.SharedPrefKeysetReader.read (SharedPrefKeysetReader.java:84)
  at com.google.crypto.tink.CleartextKeysetHandle.read (CleartextKeysetHandle.java:58)
  at com.google.crypto.tink.integration.android.AndroidKeysetManager$Builder.read (AndroidKeysetManager.java:328)
  at com.google.crypto.tink.integration.android.AndroidKeysetManager$Builder.readOrGenerateNewKeyset (AndroidKeysetManager.java:287)
  at com.google.crypto.tink.integration.android.AndroidKeysetManager$Builder.build (AndroidKeysetManager.java:238)
  at androidx.security.crypto.EncryptedSharedPreferences.create (EncryptedSharedPreferences.java:123)
  at org.openhab.habdroid.core.OpenHabApplication.getEncryptedSharedPrefs (OpenHabApplication.kt:72)
  at org.openhab.habdroid.core.OpenHabApplication.access$getEncryptedSharedPrefs (OpenHabApplication.kt:44)
  at org.openhab.habdroid.core.OpenHabApplication$secretPrefs$2.invoke (OpenHabApplication.kt:52)
  at org.openhab.habdroid.core.OpenHabApplication$secretPrefs$2.invoke (OpenHabApplication.kt:49)
  at kotlin.SynchronizedLazyImpl.getValue (LazyJVM.kt:74)
  at org.openhab.habdroid.core.OpenHabApplication.getSecretPrefs (OpenHabApplication.kt:49)
  at org.openhab.habdroid.util.ExtensionFuncsKt.getSecretPrefs (ExtensionFuncs.kt:289)
  at org.openhab.habdroid.core.connection.ConnectionFactory$Companion.initialize (ConnectionFactory.kt:523)
  at org.openhab.habdroid.core.OpenHabApplication.onCreate (OpenHabApplication.kt:90)
  at android.app.Instrumentation.callApplicationOnCreate (Instrumentation.java:1223)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:6762)
````

--- a/mobile/src/main/AndroidManifest.xml
+++ b/mobile/src/main/AndroidManifest.xml
@@ -48,6 +48,7 @@
         android:allowBackup="true"
         android:fullBackupOnly="true"
         android:fullBackupContent="@xml/backup_rules"
+        android:dataExtractionRules="@xml/backup_rules_v31"
         android:backupAgent=".util.CustomBackupAgent"
         android:icon="@mipmap/icon"
         android:label="@string/app_name"

--- a/mobile/src/main/res/xml/backup_rules_v31.xml
+++ b/mobile/src/main/res/xml/backup_rules_v31.xml
@@ -1,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<data-extraction-rules>
+    <cloud-backup disableIfNoEncryptionCapabilities="true">
+        <include domain="sharedpref" path="org.openhab.habdroid.beta_preferences.xml"/>
+        <include domain="sharedpref" path="org.openhab.habdroid_preferences.xml"/>
+    </cloud-backup>
+    <device-transfer>
+        <include domain="sharedpref" path="org.openhab.habdroid.beta_preferences.xml"/>
+        <include domain="sharedpref" path="org.openhab.habdroid_preferences.xml"/>
+    </device-transfer>
+</data-extraction-rules>

